juju/api.go

```diff
@@ -65,6 +66,12 @@ func NewAPIConnection(args NewAPIConnectionParams) (_ api.Connection, err error)
 	}
 	apiInfo, controller, err := connectionInfo(args)
 	if err != nil {
+		if errors.Is(errors.Cause(err), errors.NotValid) {
+			err = errors.NewNotValid(nil,
+				fmt.Sprintf("%v\nValid users may or may not be qualified with an @domain suffix. Examples of valid users include bob, bob@local, bob@somewhere-else, 0-a-f@123",
```

Comment:
I was thinking more along the lines of:

"A user name may contain any case alpha-numeric characters, '+', '.', and '-'; '@' to specify an optional domain. The user name and domain must begin and end with alpha-numeric characters. Examples of valid users include bob, Bob@local, bob@somewhere-else, 0-a-f@123"

Reply:
Added the text.I will have to squash the commits after.